client/components/PriceSelector/PriceSelector.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './PriceSelector.css';
+
+export default class PriceSelector extends React.Component {
+  static choices = Array(4)
+    .fill(null)
+    .map((v, i) => ({
+      id: i + 1,
+      text: Array(i + 1)
+        .fill('$')
+        .join(''),
+    }));
+
+  handleChange = (e) => {
+    const { value: currVal, onChange } = this.props;
+    const { value, checked } = e.target;
+    const newVal = {
+      ...currVal,
+      [value]: checked,
+    };
+    if (typeof onChange === 'function') onChange(newVal);
```

Comment:
so if we dont have onchange passed over is this component still valid? if it's not even valid maybe you should consider make local state out of it otherwise it will do nothing when there is no onchange

Reply:
Yes it will not be a valid component. It is my intended design since it is a controlled input component that is why the on Change is a required prop